execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -417,6 +418,831 @@ impl Distribution<ExecConfig> for Standard {
+            ),
+        );
+        named_keys.insert(UNBONDING_DELAY_KEY.into(), unbonding_delay_uref.into());
+
+        let entry_points = self.auction_entry_points();
+
+        let access_key = self
+            .uref_address_generator
+            .borrow_mut()
+            .new_uref(AccessRights::READ_ADD_WRITE);
+
+        let (_, auction_hash) = self.store_contract(access_key, named_keys, entry_points);
+
+        Ok(auction_hash)
+    }
+
+    pub(crate) fn create_standard_payment(&self) -> ContractHash {
+        let named_keys = NamedKeys::new();
+
+        let entry_points = self.standard_payment_entry_points();
+
+        let access_key = self
+            .uref_address_generator
+            .borrow_mut()
+            .new_uref(AccessRights::READ_ADD_WRITE);
+
+        let (_, standard_payment_hash) = self.store_contract(access_key, named_keys, entry_points);
+
+        standard_payment_hash
+    }
+
+    pub(crate) fn create_accounts(&self) -> Result<(), GenesisError> {
+        let accounts = {
+            let mut ret: Vec<GenesisAccount> =
+                self.exec_config.accounts().to_vec().into_iter().collect();
+            let system_account = GenesisAccount::system(Motes::zero(), Motes::zero());
+            ret.push(system_account);
+            ret
+        };
+
+        for account in accounts {
+            let account_hash = account.account_hash;
+            let main_purse = self.create_purse(
+                account.balance().value(),
+                DeployHash::new(account_hash.value()),
+            )?;
+
+            let key = Key::Account(account_hash);
+            let stored_value = StoredValue::Account(Account::create(
+                account_hash,
+                Default::default(),
+                main_purse,
+            ));
+
+            self.tracking_copy.borrow_mut().write(key, stored_value);
+        }
+
+        Ok(())
+    }
+
+    fn initial_seigniorage_recipients(
+        &self,
+        validators: &BTreeMap<PublicKey, Bid>,
+        auction_delay: u64,
+    ) -> BTreeMap<u64, SeigniorageRecipients> {
+        let initial_snapshot_range = INITIAL_ERA_ID..=INITIAL_ERA_ID + auction_delay;
+
+        let mut seigniorage_recipients = SeigniorageRecipients::new();
+        for (era_validator, founding_validator) in validators {
+            seigniorage_recipients.insert(
+                *era_validator,
+                SeigniorageRecipient::from(founding_validator),
+            );
+        }
+
+        let mut initial_seigniorage_recipients = SeigniorageRecipientsSnapshot::new();
+        for era_id in initial_snapshot_range {
+            initial_seigniorage_recipients.insert(era_id, seigniorage_recipients.clone());
+        }
+        initial_seigniorage_recipients
+    }
+
+    fn create_purse(&self, amount: U512, deploy_hash: DeployHash) -> Result<URef, GenesisError> {
+        let tracking_copy = Rc::clone(&self.tracking_copy);
```

Comment:
I wonder if it is necessary to do this...

Reply:
removed this